### DISCUSSION TITLE
Inconsistent treatment of id in roll_time_series

### DISCUSSION THREAD

**Discussion by david-waterworth** [blue-yonder/tsfresh#1000] [General]
I'm not sure if this is a bug, but the internal helper in `roll_time_series` hard codes the `id` colume regardless of the value of the input argument `column_id`

https://github.com/blue-yonder/tsfresh/blob/da06274958a10d28f96c19488af43052dc5a26ed/tsfresh/utilities/dataframe_functions.py#L346

This means that it either replaces the original id (column_id=='id'), or retains it (column_id!='id') which seems sligtly inconsistent?

**Comment by nils-braun** [comment]
That is correct @david-waterworth - thanks for spotting this.
I guess it should be replaced by `column_id` - keeping the original name makes the most sense to me. Would you like to do a PR for this?

**Comment by david-waterworth** [comment]
I started on a PR and noticed another hard-coded references, i.e. in `roll_time_series` the final results are sorted by "id" i.e.

    return df_shift.sort_values(by=["id", column_sort or "sort"])

I'm wondering if this should be

    return df_shift.sort_values(by=[column_id or "id", column_sort or "sort"])

**Comment by nils-braun** [comment]
If you are using `column_id` in the helper function, the result should only have `column_id` and not `id` anymore or did I miss something? Feel free to do the PR anyways and we can discuss more easily in the review alongside the code :-)

**Reply by nils-braun** [reply]
Ah sorry, you did already create a PR (I went through my notifications from top to bottom...) :-)
